styx-common/src/main/java/com/spotify/styx/state/RunState.java

```diff
@@ -286,7 +290,11 @@ public RunState retryAfter(WorkflowInstance workflowInstance, long delayMillis)
       switch (state()) {
         case TERMINATED:
         case FAILED:
```

Comment:
I removed the `enqueue` and `dequeue` events from this PR, so it's mainly focused on just introducing the `QUEUED` state.